### DISCUSSION TITLE
Pull the project from GitHub to Self hosted runner

### DISCUSSION THREAD

**Discussion by SatheeshRajuParayil** [community/community#46153] [New to GitHub]
Hi All,

I have a project in my GitHub repository and a self-hosted runner machine is also configured to the repository to run the project. I have created a workflow file which will run the project on the self-hosted runner. As of now I always want to log into the self-hosted runner and run a git pull command to get the latest project from the repository to a folder location in the runner machine. Is there any way to automate the process of pulling the latest code from the repository to the self-hosted runner every time when the workflow is triggered from GitHub?
Some line of commands in the same workflow file which will pull the latest project and run the project in the self-hosted runner

Thanks in advance.
Satheesh Raju

**Comment by shayan1234554321** [comment]
Yes, you can automate the process of pulling the latest code from your GitHub repository to your self-hosted runner machine by using a step in your GitHub Actions workflow.

Here is an example of how you can do this using the actions/checkout action:
yaml
`
steps:
  - name: Checkout code
    uses: actions/checkout@v2
  - name: Run project
    run: |
      cd your_project_folder
      git pull origin master
      . Add any other commands required to run the project
 `

In this example, the actions/checkout@v2 action is used to checkout the code from your GitHub repository. The second step runs a series of commands that change to the directory containing your project, run a git pull to update the code, and then run any other commands required to run the project.

You can replace your_project_folder with the actual path to your project on the self-hosted runner machine. And, replace origin master with the correct remote repository and branch, if different.

Note: This is just a basic example to illustrate the process of pulling the latest code from a GitHub repository to a self-hosted runner machine using a GitHub Actions workflow. You may need to modify it based on your specific requirements and setup.

**Reply by SatheeshRajuParayil** [reply]
Thanks @shayan1234554321
I tried the code snippet you provided with some modification as per my requirement.
But when the Git pull command is triggered am getting the error " _git : The term 'git' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the 
spelling of the name, or if a path was included, verify that the path is correct and try again._"

This is the code I used in the workflow file
![image](https://user-images.githubusercontent.com/111489222/220911775-5dd7dc06-70f3-43c6-aa80-47c40f327dc6.png)

The output am getting once executed is
![image](https://user-images.githubusercontent.com/111489222/220912017-ba8356d7-c783-4657-9f7c-9cef1905cd2a.png)

**Reply by shayan1234554321** [reply]
The error message "The term 'git' is not recognized" indicates that Git is not installed or not added to the PATH environment variable on the self-hosted runner machine 🙃 .

To fix this issue 🔴, you need to install Git on the self-hosted runner machine and add the Git executable path to the PATH environment variable. Here are the steps you can follow:

1. Install Git on the self-hosted runner machine. You can download Git from the official Git website and install it using the installer appropriate for your operating system.
2. After installing Git, check the installation by opening a command prompt and running the command git --version. If the Git installation is successful 🟢,  the command will output the version of Git installed.
3. Add the Git executable path to the PATH environment variable. On Windows, you can add the Git executable path by following these steps:

    a. Open the Control Panel and go to System > Advanced system settings > Environment Variables.

    b. Under System Variables, scroll down to find the Path variable and click Edit.

    c. Add the path to the Git executable (usually C:\Program Files\Git\bin) to the list of paths in the Variable value field. Separate multiple 
        paths with a semicolon (;).

    d. Click OK to save the changes and close the Environment Variables window.

4. After adding Git to the PATH environment variable, restart any open command prompt windows to apply the changes.

Once you have installed Git and added it to the PATH environment variable, try running the Git pull command again and it should work without any errors. 🥳
